sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -124,13 +124,27 @@ public SentryClient(final @NotNull SentryOptions options, @Nullable Connection c
     return sentryId;
   }
 
-  private @Nullable SentryEnvelope buildEnvelope(final @Nullable SentryBaseEvent event)
+  private @NotNull List<Attachment> getAttachmentsFromScope(@Nullable Scope scope) {
+    List<Attachment> attachments;
+    if (scope != null) {
+      attachments = scope.getAttachments();
+    } else {
+      attachments = new ArrayList<>();
```

Comment:
M: if the `buildEnvelope` accepts a nullable `List<Attachment>`, we'd not need to allocate this, right?

Reply:
I usually prefer passing an empty list around instead of `null`, because I don't like `null`. I could change it to `new ArrayList<>(0)` so we don't allocate too much memory, but returning `null` saves us even more memory. I'm going to change it.

Reply:
either way is fine for me, but it's not fine to return an empty instance and still, the other method accepts null and do a NPE check, ideally, either we null-safety using `new ArrayList<>(0)` and `@NotNull` for `buildEnvelope` or we just return null and let `buildEnvelope` deal with the nullability pain.
the current way is doing an extra allocation + extra NPE check for nothing.